cmd/internal/cli/verify.go

```diff
@@ -167,6 +213,72 @@ func doVerifyCmd(cmd *cobra.Command, cpath string) {
 	var opts []singularity.VerifyOpt
 
 	switch {
+	case cmd.Flag(verifyCertificateFlag.Name).Changed:
+		cert, err := loadCertificate(certificatePath)
+		if err != nil {
+			sylog.Fatalf("Failed to load certificate: %v", err)
+		}
+		opts = append(opts, singularity.OptVerifyWithCertificate(cert))
+
+		// If the certificate is self-signed certificate, we can use the certificate itself as one of the roots.
+		// Otherwise, we need the certificate of the authorities who signed the certificate.
+		var certificateIsSelfSigned bool
```

Comment:
I think it would be wise to tackle any self-signed certificate functionality separate from the online verification work. We probably need to discuss what _should_ happen when a self-signed certificate is specified. My gut instinct is that Singularity should either:
- Not accept a self-signed cert unless it is present in the system pool, or
- Accept it only if the user has specified this as the desired behaviour.

As an example of how this is handled elsewhere, `wget --no-check-certificate` allows a self-signed cert to be used. That being said, its documentation actually recommends the first approach for self-signed certs:
  > For self-signed/internal certificates, you should download the
           certificate and verify against that instead of forcing this insecure mode.